docs/developer/architecture/development-plugin-saved-objects.asciidoc

```diff
@@ -0,0 +1,205 @@
+[[development-plugin-saved-objects]]
+== Using Saved Objects
+
+Saved Objects allow {kib} plugins to use {es} like a primary
+database. Think of it as an Object Document Mapper for {es}. Once a
+plugin has registered one or more Saved Object types, the Saved Objects client
+can be used to query or perform create, read, update and delete operations on
+each type.
+
+By using Saved Objects your plugin can take advantage of the following
+features:
+ 
+* Migrations can evolve your document's schema by transforming documents and
+ensuring that the field mappings on the index are always up to date.
+* An HTTP API is automatically exposed for each type (unless the type is
+hidden).
```

Comment:
Should we add more info about these 'automatic' endpoints, or link to another doc?

Reply:
Yes, a link to the API docs will be helpful, I'll add that. I also agree that it's worth fleshing this out more in the future.